### PR TITLE
Added floor and ceiling intrinsics for `System.Runtime.Intrinsics.Vector128` and `System.Numerics.Vector`

### PR DESCRIPTION
**Pull request by LeVladIonescu**
Contributes to https://github.com/dotnet/runtime/issues/80566

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -1325,8 +1327,14 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		if (!type_enum_is_float (arg0_type))
 			return NULL;
 #ifdef TARGET_ARM64
-		int ceil_or_floor = id == SN_Ceiling ? INTRINS_AARCH64_ADV_SIMD_FRINTP : INTRINS_AARCH64_ADV_SIMD_FRINTM;
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:1328]: This could be implemented by handling OP_XOP_OVR_X_X in mini-arm64.c ?

Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:1328]: I tried like that firstly but INTRINS_AARCH64_ADV_SIMD_FRINTP and INTRINS_AARCH64_ADV_SIMD_FRINTM were not defined anywhere…I assume they are LLVM specific.

**Comment by fanyang-mono** [comment]
It is worth mentioning in the description box that this PR contributes to a certain issue.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:1328]: This is okay for now. We probably need to clean it up later, such as unify the label names.

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:1328]: Yeah, together with `llvm-intrinsics.h`. And the design only considered LLVM scenarios. Now we have llvm and non-llvm. We need to somehow make the labels/keys fit both backends.

Comment (vargaz) [reply src/mono/mono/mini/simd-intrinsics.c:1328]: A possible approach would be to use the llvm intrinsics in mini-arm64.c as well, they are just constants. If additional constants are needed to encode instructions that llvm doesn't have, they could be added after them, i.e. after INTRINS_NUM.

Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:1328]: We want to do this now or after we enable all intrinsics?

**Comment by jandupej** [comment]
Some of your CI failures should be solved by https://github.com/dotnet/runtime/pull/83770

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/mini-arm64.c [hunk 2]
```diff
@@ -21,6 +21,7 @@
 #include "ir-emit.h"
 #include "aot-runtime.h"
 #include "mini-runtime.h"
+#include "llvm-intrinsics-types.h"
```
Comment (fanyang-mono) [inline src/mono/mono/mini/mini-arm64.c:24]: This is not needed anymore, since line 31 has it already added from Jan's PR.

**Review by fanyang-mono — APPROVED** [review]

**Review by jandupej — APPROVED** [review]

**Comment by LeVladIonescu** [comment]
CI failures not related to this PR

File: src/mono/mono/mini/simd-intrinsics.c [hunk 3]
```diff
@@ -1205,10 +1205,10 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		return NULL;
 #endif
 // FIXME: This limitation could be removed once everything here are supported by mini JIT on arm64
-#ifdef TARGET_ARM64
-	if (!(cfg->compile_aot && cfg->full_aot && !cfg->interp))
-		return NULL;
-#endif
+// #ifdef TARGET_ARM64
+// 	if (!(cfg->compile_aot && cfg->full_aot && !cfg->interp))
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:1209]: This needs to be removed before merge.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 4]
```diff
@@ -1205,10 +1205,10 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		return NULL;
 #endif
 // FIXME: This limitation could be removed once everything here are supported by mini JIT on arm64
-// #ifdef TARGET_ARM64
-// 	if (!(cfg->compile_aot && cfg->full_aot && !cfg->interp))
-// 		return NULL;
-// #endif
+#ifdef TARGET_ARM64
```
Comment (SamMonoRT) [inline src/mono/mono/mini/simd-intrinsics.c]: @fanyang-mono - can you confirm if all of these are supported to remove this check ?

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c]: When all the API's are supported, we could remove this check.

**Pull request merged**